crawler/crawler/spiders/recursive_spider.py

```diff
@@ -170,14 +175,19 @@ def parse_items(self, response):
         print("Full URL: ", response.url)
         print("Depth: ", item['depth'])
         item['image_urls'] = self.collect_image_URLs(response)
-        
+
         item['file_urls'], item['file_text'] = self.collect_file_URLs(domain, item, response)
         print(item['file_urls'])
         yield item
         # Will this be recursive for all of eternity?
         if 'text/html' in str(response.headers['Content-Type']):
-            for href in response.xpath('//a/@href').getall():
-                yield Request(response.urljoin(href), self.parse_items)
+            yield from response.follow_all(
+                css="a[href]" \
+                    + ":not([href^='javascript:'])" \
+                    + ":not([href^='tel:'])" \
+                    + ":not([href^='mailto:'])",
+                callback=self.parse_items
```

Comment:
`a[href]` select all `<a>` tags with attribute `href`.
`:not([href^='javascript:'])` ensures that the href does not have a `javascript:` prefix.